lib/active_merchant/billing/gateways/ixopay.rb

```diff
@@ -3,17 +3,24 @@
 module ActiveMerchant #:nodoc:
   module Billing #:nodoc:
     class IxopayGateway < Gateway
-      self.test_url = 'https://secure.ixopay.com/transaction'
-      self.live_url = 'https://secure.ixopay.com/transaction'
+      self.test_url = 'https://secure.cardflo.io/transaction'
+      self.live_url = 'https://secure.cardflo.com/transaction'
 
       self.supported_countries = %w(AO AQ AR AS AT AU AW AX AZ BA BB BD BE BF BG BH BI BJ BL BM BN BO BQ BQ BR BS BT BV BW BY BZ CA CC CD CF CG CH CI CK CL CM CN CO CR CU CV CW CX CY CZ DE DJ DK DM DO DZ EC EE EG EH ER ES ET FI FJ FK FM FO FR GA GB GD GE GF GG GH GI GL GM GN GP GQ GR GS GT GU GW GY HK HM HN HR HT HU ID IE IL IM IN IO IQ IR IS IT JE JM JO JP KE KG KH KI KM KN KP KR KW KY KZ LA LB LC LI LK LR LS LT LU LV LY MA MC MD ME MF MG MH MK ML MM MN MO MP MQ MR MS MT MU MV MW MX MY MZ NA NC NE NF NG NI NL NO NP NR NU NZ OM PA PE PF PG PH PK PL PM PN PR PS PT PW PY QA RE RO RS RU RW SA SB SC SD SE SG SH SI SJ SK SL SM SN SO SR SS ST SV SX SY SZ TC TD TF TG TH TJ TK TL TM TN TO TR TT TV TW TZ UA UG UM US UY UZ VA VC VE VG VI VN VU WF WS YE YT ZA ZM ZW)
       self.default_currency = 'EUR'
       self.currencies_with_three_decimal_places = %w(BHD IQD JOD KWD LWD OMR TND)
-      self.supported_cardtypes = %i[visa master american_express discover diners_club jcb maestro]
+      self.supported_cardtypes = %i[visa master american_express discover diners_club]
 
       self.homepage_url = 'https://www.ixopay.com'
       self.display_name = 'Ixopay'
 
+      SUCCESS_CODE = '111111111' # TODO check once API works
+
+      # there aren't soft and hard error categories in the docs
+      # 2022 - 3D-Secure Soft declined code
+      # TODO check other code in the docs https://gateway.cardflo.io/documentation/gateway#error-codes
+      SOFT_DECLINE_CODES = %w{2022}.freeze
```

Comment:
Hi @ruslan-nastasiev , We need to check which codes are `Soft Decline` and `Hard decline`, In other payment gateways, we have a similar kind of error codes, and we identified which are soft decline ones and which are hard declines.

Reply:
**Soft Decline errors :-**
1. Do not honor.
2. Over limit.
3. Transaction not allowed.
4. No such card issuer.
5. No card number on file with issuer.
6. Invalid expiration date.
7. Invalid card security code.
8. Invalid PIN.
9. Call issuer for further information.

Rest all errors are **hard decline** errors.

Reply:
Please include 3ds errors as well, which are soft decline ones.